server/routes/api/import_detection/detection_types.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Classes (Enum, dataclass) used in the the various import_detection modules."""
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Sequence, Union
+
+
+# Helper Enums and Data Classes for Detection.
+class MappingType(Enum):
+    """Supported types of column structure."""
+    COLUMN = "column"
+    COLUMN_HEADER = "columnHeader"
+
+
+class MappedThing(Enum):
+    """Supported type of Thing being mapped."""
+    DATE = "Date"
+    PLACE = "Place"
+
+
+@dataclass
+class Column:
+    """Data Type for a Column."""
+    # The Id of the column.
+    id: str
+
+    # Column header name.
+    header: str
+
+    # Column index (starting with 0).
+    column_index: int
+
+
+@dataclass
+class Entity:
+    """A DC Entity."""
+    dcid: str
+    display_name: str
+
+
+class DCType(Entity):
+    """A Data Commons entity type, e.g. Country (which is a type of Place)."""
+
+
+class DCProperty(Entity):
+    """A Data Commons property, e.g. longitude."""
+
+
+@dataclass
+class TypeProperty:
+    """A combination of the DC Type and associated Property."""
+    dc_type: DCType
+    dc_property: DCProperty
+
+
+@dataclass
+class MappingVal:
+    # The value of the MappingType enum (column or columnHeader)
+    type: str
+
+    # Column that holds the mapping values. Should be set if type is MappingType.COLUMN
+    column: Union[Column, None]
```

Comment:
For this pattern, maybe we can use Optional? https://docs.python.org/3/library/typing.html#typing.Optional

Reply:
Oh yes, true! Done. Thanks :)